dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -337,19 +337,20 @@ private static DependencyGraph levelOrder(
   }
 
   /**
-   * Returns true if {@code exceptionAndPath.getPath} does not contain {@code optional} dependency
-   * and the path does not contain {@code scope:provided} dependency.
+   * Returns true if {@code dependencyPath} does not contain {@code optional} dependency and the
+   * path does not contain {@code scope:provided} dependency.
    */
-  private static boolean isUnacceptableResolutionException(ExceptionAndPath exceptionAndPath) {
-    ImmutableList<DependencyNode> dependencyNodes = exceptionAndPath.getPath();
+  public static boolean unacceptableMissingDependency(List<DependencyNode> dependencyPath) {
     boolean hasOptionalParent =
```

Comment:
I'm not sure this variable name is correct. You're checking the entire path, not one parent, right?

Reply:
Yes, removed `parent`.

"Dependency" attribute is relation from parent to child. If a node has `provided` scope, then it means parent specified it as "provided".